cordova-lib/src/cordova/platform.js

```diff
@@ -110,15 +110,25 @@ function addHelper(cmd, hooksRunner, projectRoot, targets, opts) {
                 if(platform === 'wp8') {
                     events.emit('warn', 'wp8 has been deprecated. Please use windows instead.');
                 }
-                if(spec && pkgJson) {
-                    pkgJson.dependencies[pkgJsonSpec] = spec;
+                // If there is a spec specified during add, add to pkg.json dependencies(if exists).
```

Comment:
Thank you! So first I tried this, but if you just check if they satisfy each other than it will never re-write to package.json.

```                 if(spec && pkgJson && pkgJson.dependencies && (pkgJson.dependencies[prefixCordovaPlatform] || pkgJson.dependencies[platform])) {
                    if ((semver.satisfies(spec, pkgJson.dependencies[prefixCordovaPlatform])) || (semver.satisfies(spec, pkgJson.dependencies[platform]))) {
                    } else {
                        if (pkgJson.dependencies[prefixCordovaPlatform]) {
                            pkgJson.dependencies[prefixCordovaPlatform] = '^'+spec;
                        } else if (pkgJson.dependencies[platform]) {
                            pkgJson.dependencies[platform] = '^'+spec;
                        }
                        fs.writeFileSync(pkgJsonPath, JSON.stringify(pkgJson, null, 4), 'utf8');
                    }
                } ```